core/src/main/java/io/questdb/std/str/Utf8Sequence.java

```diff
@@ -51,8 +51,47 @@ public interface Utf8Sequence {
     byte byteAt(int index);
 
     /**
-     * @return true if all characters in the string are ASCII. This is a best-effort flag, i.e. it may have
-     * false value while the actual string has ASCII characters only.
+     * Returns 8 bytes of content packed into a long (little-endian).
+     * Bytes beyond the end of the string are 0.
+     */
+    default long zeroPaddedLongAt(int index) {
```

Comment:
It could be a better idea to add unsafe `longAt` (and maybe `intAt`) methods to this interface. We already have them implemented in `Utf8String`. Then `equalsAssumingSameSize` could first check `zeroPaddedSixPrefix` and then start a loop based on long/int/byte comparison similar to `Vect#memeq0(long, long, long)`.

Nice side effect of having these methods is that we can use them in `MapKey#putVarchar` and other similar places where we currently have per-byte access to the string contents.

Reply:
The fastest implementation would rely on zero-padded long, with no need for special-cased tail code. Don't you think?

Reply:
> The fastest implementation would rely on zero-padded long, with no need for special-cased tail code. Don't you think?

Zero-padded long implementation already assumes special-cased tail code or redundant zeroed memory.

Reply:
The redundant memory is already there in the inlined varchar, which was my rationale for having this method. But I also think it would be a great idea to add a few bytes to the end of a varchar data vector, so we can have an optimal implementation there as well, with no tail code.